go/store/types/chunkedstring.go

```diff
@@ -0,0 +1,332 @@
+// Copyright 2020 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"context"
+	"errors"
+	"math"
+)
+
+type ChunkedString struct {
+	valueImpl
+}
+
+var _ Value = ChunkedString{}
+var _ valueReadWriter = ChunkedString{}
+
+//TODO: Consolidate all chunk size literals into global constants (this should be the smallest constant)
+const maxChunkedStringChunk = 1<<27 - 1024
```

Comment:
Discussion of how this number is derived?

Reply:
From searching around the codebase to see what the lowest number for any table or chunk size limit that I could find, and then subtracting `1024` from it to account for prepended metadata (such as noms kinds, etc.). So the `1024` is arbitrary and can honestly be like `4` bytes, but it doesn't matter too much.

However, all over the code base we've got some places referencing a constant, and other places are just `1<<28`, so it's hard to search for them all, so I added the `TODO` so we can just have a collection of constants to reference.